src/NLog/Internal/PropertiesDictionary.cs

```diff
@@ -0,0 +1,744 @@
+            }
+            return false;
+        }
+
+        /// <inheritDoc/>
+        public bool Remove(KeyValuePair<object, object> item)
+        {
+            if (!IsEmpty)
+            {
+                if (((IDictionary<object, PropertyValue>)EventProperties).Remove(new KeyValuePair<object, PropertyValue>(item.Key, new PropertyValue(item.Value, false))))
+                    return true;
+
+                if (((IDictionary<object, PropertyValue>)EventProperties).Remove(new KeyValuePair<object, PropertyValue>(item.Key, new PropertyValue(item.Value, true))))
+                    return true;
+            }
+
+            return false;
+        }
+
+        /// <inheritDoc/>
+        public bool TryGetValue(object key, out object value)
+        {
+            if (!IsEmpty)
+            {
+                PropertyValue valueItem;
+                if (EventProperties.TryGetValue(key, out valueItem))
+                {
+                    value = valueItem.Value;
+                    return true;
+                }
+            }
+
+            value = null;
+            return false;
+        }
+
+        /// <inheritDoc/>
+        bool IDictionary.Contains(object key)
+        {
+            return ContainsKey(key);
+        }
+
+        /// <inheritDoc/>
+        void IDictionary.Remove(object key)
+        {
+            Remove(key);
+        }
+
+        /// <inheritDoc/>
+        void ICollection.CopyTo(Array array, int arrayIndex)
+        {
+            if (array == null)
+                throw new ArgumentNullException("array");
+            if (arrayIndex < 0)
+                throw new ArgumentOutOfRangeException("arrayIndex");
+
+            if (!IsEmpty)
+            {
+                foreach (var propertyItem in this)
+                {
+                    array.SetValue(new DictionaryEntry(propertyItem.Key, propertyItem.Value), arrayIndex);
+                    ++arrayIndex;
+                }
+            }
+        }
+
+        void ResetMessageProperties()
+        {
+            if (_eventProperties != null && _messageProperties != null && _messageProperties.Count > 0)
+            {
+                string key;
+                PropertyValue valueItem;
+                for (int i = 0; i < _messageProperties.Count; ++i)
+                {
+                    key = _messageProperties[i].Name;
+                    if (_eventProperties.TryGetValue(key, out valueItem))
+                    {
+                        if (valueItem.MessageProperty)
+                        {
+                            _eventProperties.Remove(key);
+                        }
+                    }
+                }
+                _messageProperties = ArrayHelper.Empty<MessageTemplateParameter>();
+            }
+        }
+
+        private static bool InsertMessagePropertiesDictionary(IList<MessageTemplateParameter> messageProperties, Dictionary<object, PropertyValue> eventProperties)
```

Comment:
for the docs phase: please document bool return values

Reply:
Please wait with doc-stuff until we have something real